math/n_bonacci.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ * @brief Implementation of [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers) series
+ *
+ * @details
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <algorithm>  /// for std::is_equal, std::swap
+#include <cassert>    /// for assert
+#include <iostream>   /// for io operations
+#include <vector>     /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+ * @namespace n_bonacci
+ * @brief Functions for the [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers) implementation
+ */
+namespace n_bonacci {
+/**
+ * @brief Function implements n-bonacci series
+ * @tparam T type of array
+ * @param n,m m is the number of terms in n-bonacci sequence
```

Comment:
The `m` parameter is missing documentation here though (I think).
```suggestion
 * @param n is the number of terms in the N-Bonacci sequence
```